test-poc/framework/src/main/java/org/keycloak/test/framework/injection/Registry.java

```diff
@@ -209,10 +211,47 @@ private InstanceWrapper getDeployedInstance(Supplier supplier) {
     }
 
     private void loadSuppliers() {
-        ServiceLoader.load(Supplier.class).iterator().forEachRemaining(suppliers::add);
+        Iterator<Supplier> supplierIterator = ServiceLoader.load(Supplier.class).iterator();
+        Set<Class> loadedValueTypes = new HashSet<>();
+        Set<Supplier> skippedSuppliers = new HashSet<>();
+
+        while (supplierIterator.hasNext()) {
+            Supplier supplier = supplierIterator.next();
```

Comment:
`supplier.getValueType()` is  called 4 times in this loop, perhaps re-use it in a variable?

Reply:
Updated